symengine/functions.cpp

```diff
@@ -2895,6 +2895,159 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
     return make_rcp<const UpperGamma>(s, x);
 }
 
+Beta::Beta(const RCP<const Basic> &x, const RCP<const Basic> &y)
+    : x_{x}, y_{y}
+{
+    SYMENGINE_ASSERT(is_canonical(x_, y_))
+}
+
+bool Beta::is_canonical(const RCP<const Basic> &x, const RCP<const Basic> &y)
+{
+    if (is_a<Integer>(*x) or (is_a<Rational>(*x) and
+        (rcp_static_cast<const Rational>(x)->i.get_den()) == 2)) 
+    {
+        if (is_a<Integer>(*y) or (is_a<Rational>(*y) and
+        (rcp_static_cast<const Rational>(y)->i.get_den()) == 2)) {
+            return false;
+        }
+    }
+    return false;
+}
+
+std::size_t Beta::__hash__() const
+{
+    std::size_t seed = BETA;
+    hash_combine<Basic>(seed, *x_);
+    hash_combine<Basic>(seed, *y_);
+    return seed;
+}
+
+bool Beta::__eq__(const Basic &o) const
+{
+    if (is_a<Beta>(o) and
+        eq(*x_, *(static_cast<const Beta &>(o).x_)) and
+        eq(*y_, *(static_cast<const Beta &>(o).y_)))
```

Comment:
Since `beta` is symmetric, is it necessary to enforce the order of the arguments here?